### PR TITLE
Fixes #898: Replace hashicorp and terraform references

### LINKED ISSUE #898: Swap `hashicorp`, or `terraform` CLI identification occurrences in the codebase with `OpenTofu`
## Problem

The codebase contains several places where `tofu` is identified as `terraform`. Also, `app.hashicorp.io` is used as URL for tests, and URL.

Examples:
- [cmd/tofu/telemetry.go](https://github.com/opentofu/opentofu/blob/d077a33939ddbe07f8a71d75d8d61c127fee6cd1/cmd/tofu/telemetry.go#L65)
- [internal/backend/remote/testing.go](https://github.com/opentofu/opentofu/blob/d077a33939ddbe07f8a71d75d8d61c127fee6cd1/internal/backend/remote/testing.go#L73-L74)
- [internal/cloud/backend.go](https://github.com/opentofu/opentofu/blob/d077a33939ddbe07f8a71d75d8d61c127fee6cd1/internal/cloud/backend.go#L43-L46)
- [internal/registry/regsrc/friendly_host_test.go](https://github.com/opentofu/opentofu/blob/d077a33939ddbe07f8a71d75d8d61c127fee6cd1/internal/registry/regsrc/friendly_host_test.go#L36-L43)

**Note** please do no change occurrences mentioned in #530, for example:
- `terraform.tfstate.d`
- `terraform.tfstate`
- `X-Terraform-Get`
- Please do not change [internal/version/version.go](https://github.com/opentofu/opentofu/blob/d077a33939ddbe07f8a71d75d8d61c127fee6cd1/version/version.go#L54)

## Proposed solution

- Replace terraform occurrences with `OpenTofu`
- Fix the links to the OpenTofu documentation (see #891)

### PR DESCRIPTION
**Pull request by janosdebugs**
This PR replaces most references to Hashicorp and Terraform with OpenTofu where relevant.

Resolves #898

## Target Release

1.6.0

## Current status

Most of the tests pass, however the following tests don't and need to be fixed.

### `github.com/opentofu/opentofu/internal/backend/remote`

- [x] `TestRemote_config`
- [x] `TestRemote_VerifyWorkspaceTerraformVersion_ignoreFlagSet`
- [x] `TestRemote_ServiceDiscoveryAliases`

### `github.com/opentofu/opentofu/internal/cloud`

- [x] `TestCloud_setUnavailableTerraformVersion`
- [x] `TestCloud_localBackend`
- [x] `TestCloud_addAndRemoveWorkspacesDefault`
- [x] `TestCloud_StateMgr_versionCheck`
- [x] `TestCloud_StateMgr_versionCheckLatest`
- [x] `TestCloud_VerifyWorkspaceTerraformVersion`
- [x] `TestCloud_VerifyWorkspaceTerraformVersion_workspaceErrors`
- [x] `TestCloud_VerifyWorkspaceTerraformVersion_ignoreFlagSet`
- [x] `TestCloudBackend_DeleteWorkspace_SafeAndForce`
- [x] `TestCloudBackend_DeleteWorkspace_DoesNotExist`
- [x] `TestCloud_ServiceDiscoveryAliases`
- [x] `TestState_GetRootOutputValues`
- [x] `TestState`
- [x] `TestCloudLocks`
- [x] `TestDelete_SafeDeleteNotSupported`
- [x] `TestDelete_ForceDelete`
- [x] `TestDelete_SafeDelete`
- [x] `TestState_PersistState`
- [x] `TestState_ShouldPersistIntermediateState`

### `github.com/opentofu/internal/command`

- [x] `TestInit_providerSource`
- [x] `TestInit_getUpgradePlugins`
- [x] `TestInit_providerLockFile`
- [x] `TestInit_providerLockFileReadonly`
- [x] `TestInit_pluginDirProviders`
- [x] `TestInit_testsWithProvider`
- [x] `TestProvidersLock`
- [x] `TestTest_ValidatesBeforeExecution`

### `github.com/opentofu/opentofu/internal/configs`

- [x] `TestBuildConfigInvalidModules`
- [x] `TestLoadModuleCall`

### `github.com/opentofu/opentofu/internal/getproviders`

- [x] `TestFilesystemMirrorSourceAllAvailablePackages`
- [x] `TestFilesystemMirrorSourceAvailableVersions`
- [x] `TestFilesystemMirrorSourcePackageMeta`
- [x] `TestMultiSourceSelector`
- [x] `TestPackageHashAuthentication_success`
- [x] `TestPackageHashAuthentication_failure`
- [x] `TestArchiveChecksumAuthentication_success`
- [x] `TestArchiveChecksumAuthentication_failure`

### `github.com/opentofu/opentofu/internal/providercache`

- [x] `TestCachedProviderHash`
- [x] `TestExecutableFile`
- [x] `TestLinkFromOtherCache`
- [x] `TestDirReading`
- [x] `TestEnsureProviderVersions_local_source`

### CODE REVIEW

File: internal/addrs/provider.go [hunk 1]
```diff
@@ -14,6 +14,10 @@ import (
 // extended to include additional fields including Namespace and SourceHost
 type Provider = tfaddr.Provider
 
+// DefaultProviderNamespace contains the registry namespace OpenTofu looks up providers in
+// if no namespace has been explicitly specified.
+const DefaultProviderNamespace = "opentofu"
```
Comment (cam72cam) [inline internal/addrs/provider.go:19]: This has some interesting opentofu/registry implications

Comment (cam72cam) [reply internal/addrs/provider.go:19]: We will also need to test to make sure this does not break state files.

Comment (cam72cam) [reply internal/addrs/provider.go:19]: We did some testing in our dev sync and realized that this will not likely be a problem.

File: internal/configs/module_call_test.go [hunk 2]
```diff
@@ -50,17 +50,17 @@ func TestLoadModuleCall(t *testing.T) {
 			SourceAddr: addrs.ModuleSourceRegistry{
 				Package: addrs.ModuleRegistryPackage{
 					Host:         addrs.DefaultModuleRegistryHost,
-					Namespace:    "hashicorp",
+					Namespace:    addrs.DefaultProviderNamespace,
```
Comment (cam72cam) [inline internal/configs/module_call_test.go:53]: DefaultModuleNamespace?

File: internal/getproviders/testdata/filesystem-mirror/registry.opentofu.org/hashicorp/null/2.0.0/windows_amd64/terraform-provider-null.exe [hunk 3]
```diff
@@ -1 +0,0 @@
-# This is just a placeholder file for discovery testing, not a real provider plugin.
```
Comment (cam72cam) [inline internal/getproviders/testdata/filesystem-mirror/registry.opentofu.org/hashicorp/null/2.0.0/windows_amd64/terraform-provider-null.exe:1]: Why was this removed?

File: internal/providercache/testdata/cachedir/registry.opentofu.org/hashicorp/null/2.0.0/windows_amd64/terraform-provider-null.exe [hunk 4]
```diff
@@ -1 +0,0 @@
-# This is just a placeholder file for discovery testing, not a real provider plugin.
```
Comment (cam72cam) [inline internal/providercache/testdata/cachedir/registry.opentofu.org/hashicorp/null/2.0.0/windows_amd64/terraform-provider-null.exe:1]: Why was this removed?

File: internal/tofumigrate/tofumigrate.go [hunk 5]
```diff
@@ -53,6 +54,9 @@ func MigrateStateProviderAddresses(config *configs.Config, state *states.State)
 			if resource.ProviderConfig.Provider.Hostname == "registry.terraform.io" && !referencedInConfig {
 				resource.ProviderConfig.Provider.Hostname = tfaddr.DefaultProviderRegistryHost
 			}
+			if resource.ProviderConfig.Provider.Namespace == "hashicorp" && !referencedInConfig {
```
Comment (cam72cam) [inline internal/tofumigrate/tofumigrate.go:57]: From discover during our ad-hoc testing from yesterday, is this still needed?

File: internal/configs/module_call_test.go (hunk 2 above)
Comment (janosdebugs) [reply internal/configs/module_call_test.go:53]: This shouldn't actually change because modules shouldn't change?

File: internal/getproviders/testdata/filesystem-mirror/registry.opentofu.org/hashicorp/null/2.0.0/windows_amd64/terraform-provider-null.exe (hunk 3 above)
Comment (janosdebugs) [reply internal/getproviders/testdata/filesystem-mirror/registry.opentofu.org/hashicorp/null/2.0.0/windows_amd64/terraform-provider-null.exe:1]: It was renamed to opentofu/null/...

File: internal/tofumigrate/tofumigrate.go (hunk 5 above)
Comment (cube2222) [reply internal/tofumigrate/tofumigrate.go:57]: This is not needed, let's roll this file back.

**Review by cube2222 — APPROVED** [review]

**Review by cube2222 — APPROVED** [review]

**Review by cam72cam — APPROVED** [review]
